### PR TITLE
Update extract configs

### PR DESCRIPTION
**Pull request by lurch**
Update config-extraction scripts to work with multiple chips
Associated fixes to the config-descriptions

### CODE REVIEW

**Comment by lurch** [comment]
Ooops broken the build, so I've converted to draft while I investigate.

**Comment by lurch** [comment]
> Ooops broken the build, so I've converted to draft while I investigate.

Fixed!

File: src/rp2040/hardware_regs/include/hardware/platform_defs.h [hunk 1]
```diff
@@ -46,7 +46,7 @@
 
 #define HAS_SIO_DIVIDER 1
 #define HAS_RP2040_RTC  1
-// PICO_CONFIG: XOSC_HZ, Crystal oscillator frequency in Hz, type=int, default=12000000, advanced=true, group=hardware_base
+// PICO_CONFIG: XOSC_HZ, The crystal oscillator frequency in Hz, type=int, default=12000000, advanced=true, group=hardware_base
```
Comment (kilograham) [inline src/rp2040/hardware_regs/include/hardware/platform_defs.h:49]: Seems like we wre removing "The" in general - i assume you did this to match RP2350 perhaps, but we should probably change that instead

Comment (lurch) [reply src/rp2040/hardware_regs/include/hardware/platform_defs.h:49]: Haha. When there were descriptions that only differed slightly, I wasn't sure which direction to go in, so I assumed that as RP2350 was the "most recently edited" version, then that was therefore "the most correct". But sure, I'll tack another commit on the end here to get rid of the "The..." versions :slightly_smiling_face:

**Comment by kilograham** [comment]
were you going to add a commit here, or separately?

**Comment by lurch** [comment]
> were you going to add a commit here, or separately?

Up to you. You can either merge this PR as-is, so that I can then slide in my databooks PR that adds the tagging to the tables in chapters 5 and 6 of the [Pico-SDK PDF](https://datasheets.raspberrypi.com/pico/raspberry-pi-pico-c-sdk.pdf), or you can leave this PR open until I've done the additional work to also tackle `PICO_DEFINE` (although I dunno when that'll happen yet, as I've been busy with other stuff).

**Comment by lurch** [comment]
> were you going to add a commit here, or separately?

Now done - sorry I kept forgetting! :man_facepalming:

**Review by kilograham — APPROVED** [review]

**Pull request merged**